imap_processing/utils.py

```diff
@@ -299,9 +299,11 @@ def _create_minimum_dtype_array(values: list, dtype: str) -> np.ndarray:
         The array of values.
     """
     try:
-        return np.array(values, dtype=dtype)
+        dtype_array: np.ndarray = np.array(values, dtype=dtype)
+        return dtype_array
     except ValueError:
-        return np.array(values)
+        regular_array: np.ndarray = np.array(values)
+        return regular_array
```

Comment:
I think that for all of the numpy typing we need to import their typing module, and then use `npt.NDArray`. Saying that we are returning `np.ndarray` seems incorrect, so I think we should change those, rather than add these explicit type annotations throughout the code.

```
import numpy.typing as npt

def func() -> npt.NDArray
```

https://stackoverflow.com/questions/66349242/specific-type-annotation-for-numpy-ndarray-using-mypy